packages/react/src/editor/BlockNoteTheme.ts

```diff
@@ -29,467 +26,181 @@ export type ColorScheme = {
   };
 };
 
-export type ComponentStyles = Partial<{
-  // Slash Menu, Formatting Toolbar dropdown, color picker dropdown
-  Menu: CSSObject;
-  // Icon in the color picker dropdown (Formatting Toolbar & Drag Handle Menu)
-  ColorIcon: CSSObject;
-  DragHandleMenu: CSSObject;
-  // Menu to edit hyperlinks (in Formatting Toolbar & Hyperlink Toolbar)
-  EditHyperlinkMenu: CSSObject;
-  Editor: CSSObject;
-  // Used in the Image Toolbar
-  FileInput: CSSObject;
-  // Handle that appears next to tables and the menu that opens when clicking it
-  TableHandle: CSSObject;
-  TableHandleMenu: CSSObject;
-  // Used in the Image Toolbar
-  Tabs: CSSObject;
-  TextInput: CSSObject;
-  // Wraps Formatting Toolbar & Hyperlink Toolbar
-  Toolbar: CSSObject;
-  // Appears on hover for Formatting Toolbar
-  // & Hyperlink Toolbar buttons
-  Tooltip: CSSObject;
-  SlashMenu: CSSObject;
-  SideMenu: CSSObject;
-}>;
-
 export type Theme = {
   colors: ColorScheme;
   borderRadius: number;
   fontFamily: string;
-  componentStyles?: (theme: Theme) => ComponentStyles;
 };
 
-export const blockNoteToMantineTheme = (theme: Theme): MantineThemeOverride => {
-  const shadow = `0 4px 12px ${theme.colors.shadow}`;
-  const lightShadow = `0 2px 6px ${theme.colors.border}`;
-  const border = `1px solid ${theme.colors.border}`;
+export const themeToCSSVariables = (theme: Theme) => {
```

Comment:
should we use this manual code, or reuse sth like https://mantine.dev/styles/css-variables/#css-variables-resolver?

Reply:
I think it's better to do it manually in this case, since the theme we use is different to the Mantine theme that the `CSSVariablesResolver` uses. I think either way the amount of code is roughly the same, but if we use the `CSSVariablesResolver`, more of it becomes coupled to Mantine so it would be more of a pain if we were to switch UI libraries.

Reply:
makes sense. Should we also add code to remove the variables?

Reply:
Hmm not sure how useful that would be, or do you mean like reverting them to defaults?